treeherder/model/data_cycling/signature_remover.py

```diff
@@ -57,6 +57,10 @@ def remove_in_chunks(self, signatures):
             if emails_sent < self._max_emails_allowed and (
                 not perf_signature.has_performance_data()
             ):
+                if perf_signature.repository.name == 'try':
```

Comment:
This is still conditioned by our email thresholds.
As a result, the execution flow isn't deterministic.